mozilla_schema_generator/generic_ping.py

```diff
@@ -94,12 +94,15 @@ def make_schemas(env: Schema, probes: List[Probe], config: Config,
             final_schema.set_schema_elem(
                 schema_key + ("properties", probe.name),
                 probe_schema.schema)
-            final_schema.set_schema_elem(
-                schema_key + ("additionalProperties",),
-                False)
-            final_schema.delete_group_from_schema(schema_key + ("propertyNames",))
 
-        return schemas + [final_schema]
+        # Remove all additionalProperties (#22)
+        schemas.append(final_schema)
+        for s in schemas:
+            for key in config.get_match_keys():
+                s.delete_group_from_schema(key + ("propertyNames",), propogate=False)
+                s.set_schema_elem(key + ("additionalProperties",), False, propogate=False)
```

Comment:
Does this do the intended thing? It seems like you're removing the properties and leaving behind an additionalProperties field.

If I'm understanding correctly, this results in something like this:

```json
{
    "metrics": {
        "properties": {
            "my-probe": {
                "additionalProperties": false
            }
        }
    }
}
```

assuming a single probe in the config. This new bit of code is to avoid the following when probes are missing?

```json
{
    "metrics": {
        "additionalProperties": false
    }
}
```